### ISSUE TITLE
Allow dynamic reconfiguration

### ISSUE THREAD

**Issue by tilsche** [metricq/metricq-source-ipmi#14]
The current code does not seem to support reconfiguration, i.e. the source will not behave properly if `_on_config` is called more than once. A particular issue is that the actual data collection tasks are scheduled in `_on_config` with `asyncio.gather`, but the returned awaitable is not stored anywhere for a clean shutdown:

https://github.com/metricq/metricq-source-ipmi/blob/master/metricq_source_ipmi/source/main.py#L429

Basically, the state of the source needs to be cleaned thoroughly for a second invocation of `_on_config`.

@phijor  can you take a look at this?

When refactoring, keep performance in mind.
